app/controllers/state_file/questions/canceled_data_transfer_controller.rb

```diff
@@ -36,11 +36,8 @@ def redirect_if_no_us_state_in_params
         return if params[:us_state]
 
         if session[:state_file_intake].present?
-          if current_intake.class == StateFileAzIntake
-            redirect_link = az_questions_canceled_data_transfer_path(us_state: "az")
-          elsif current_intake.class == StateFileNyIntake
-            redirect_link = ny_questions_canceled_data_transfer_path(us_state: "ny")
-          end
+          state_code = StateFile::StateInformationService.state_code_from_intake_class(current_intake)
```

Comment:
`StateFileBaseIntake` does still implement `self.state_code`, so I think it would be better to use `current_intake.state_code` and inline it in the line below

Reply:
Perhaps `state_code_from_intake_class` could be made private to enforce using `current_intake.state_code` instead